repository_service_tuf_worker/repository.py

```diff
@@ -1212,3 +1308,91 @@ def metadata_rotation(
         logging.warn(deprecation_message)
 
         return self.metadata_update(payload, update_state)
+
+    def _sign_root(
+        self, root: Metadata[Root], sig_payload: Dict
+    ) -> Tuple[bool, str]:
+        """
+        Sign the root metadata and call actions depending on the bootstrap
+        state, and if it matches the threshold
+        """
+        signature: Signature = Signature.from_dict(sig_payload)
+        root.signatures[signature.keyid] = signature
+
+        signed = self._validate_signatures(root, Root.type)
+        bootstrap = self._settings.get_fresh("BOOTSTRAP")
+
+        if "signing-" in bootstrap:
+            if signed:
+                self._persist(root, Root.type)
+                self.write_repository_settings("ROOT_SIGNING", None)
+                bootstrap_task_id = bootstrap.split("signing-")[1]
+                self._bootstrap_online_roles(root)
+                self.write_repository_settings("BOOTSTRAP", bootstrap_task_id)
+                message = "Bootstrap finished"
+                logging.info(message)
+            else:
+                self.write_repository_settings(
+                    f"{Root.type.upper()}_SIGNING", root.to_dict()
+                )
+                message = f"Root version {root.signed.version} pending sign."
+                logging.info(message)
+
+        else:
+            message = "Signing Metadata Update not implemented (issue #336)"
+
+        return signed, message
+
+    def sign_metadata(
+        self,
+        payload: Dict[str, Dict[str, str]],
+        update_state: Optional[
+            Task.update_state
+        ] = None,  # It is required (see: app.py)
+    ) -> Dict[str, Any]:
+        """
+        Start the signing metadata process
+        """
+        rolename = payload["role"]
+        sig_payload = payload["signature"]
+
+        available_signing = list(
+            filter(lambda var: "SIGNING" in var, dir(self._settings))
```

Comment:
Why do we need this AND `self._settings.get_fresh(f"{rolename.upper()}_SIGNING")`?

Reply:
Also, would it be enough to call `get_fresh` only once? Currently, you call it twice, in L1365 and L1378.